### PR TITLE
Deprecate torch.cross default behaviour

### PR DESCRIPTION
**Pull request by lezcano**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #108760

Long overdue this one. We may be able to change it in a few years :hopeful:.

**BC-breaking note**

This PR deprecates `torch.cross`'s default dim in favor of
`torch.linalg.cross`.
A upgrade guide is added to the documentation for `torch.cross`.

Note this PR DOES NOT remove `torch.cross`.

Fixes https://github.com/pytorch/pytorch/issues/108664

cc @jianyuh @nikitaved @pearu @mruberry @walterddr @IvanYashchuk @xwang233 @Lezcano

### CODE REVIEW

**Comment by vadimkantorov** [comment]
@lezcano Please also remove default-dim from softmax :)
- https://github.com/pytorch/pytorch/issues/84524
- https://github.com/pytorch/pytorch/issues/36524

**Comment by lezcano** [comment]
That Softmax deprecation may be more painful, and it'll need to be done by someone at meta. Feel free to poke them in that other issue.

File: aten/src/ATen/native/Cross.cpp [hunk 1]
```diff
@@ -59,6 +59,13 @@ static int64_t _default_cross_dim(const c10::optional<int64_t> &dimension, SymIn
 }
 
 Tensor cross(const Tensor & input, const Tensor & other, const c10::optional<int64_t> dimension) {
+  if (!dimension) {
+    TORCH_WARN_ONCE(
+      "Using torch.cross without specifying the dim arg is deprecated.\n",
+      "Please either pass the dim explicitly or simply use linalg.cross.\n",
```
Comment (albanD) [inline aten/src/ATen/native/Cross.cpp:65]: nit:
```suggestion
      "Please either pass the dim explicitly or simply use torch.linalg.cross.\n",
```

**Review by albanD — APPROVED** [review]
nit on the doc but sounds good otherwise
